meta-cube/recipes-support/overc-system-agent/files/overc-system-agent-1.2/Overc/overc.py

```diff
@@ -68,10 +69,30 @@ def _system_upgrade(self, template, reboot, force):
                             print "*** Failed to upgrade container %s" % cn
                             print "*** Abort the system upgrade action"
                             sys.exit(self.retval)
+                        else:
+                            if self.container.pre_upgrade(cn, template) == 1:
+                                overlay_flag = 1
                         break
 
         self._host_upgrade(reboot, force)
 
+        if overlay_flag == 1:
+            # Enable lxc-overlay service in essential
+            lxc = open("/essential/lib/systemd/system/lxc.service", 'r')
```

Comment:
How are we supporting the mode that says "I'm ok with duplicate files" ? If we do want to support that, then there should be a way to override this scanning on the start.